pkg/core/plugin.go

```diff
@@ -0,0 +1,30 @@
+package core
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+)
+
+var plugins = make(map[string]Plugin)
+
+func Register(name string, plugin Plugin) {
+	if plugin == nil {
+		log.Fatal("Undefined plugin")
+	}
+	if _, isDup := plugins[name]; isDup {
+		log.Fatal("Plugin with this name already exists")
+	}
+	plugins[name] = plugin
+}
+
+func GetRootCmd(pluginName string) (*cobra.Command, error) {
```

Comment:
I'm adversary for and developer-facing API that bases on unstructured content - Plugin name as string has no way to get validation on compile time. This is huge bummer because without this we can bring plugin root command and get it validated against cobra at compile time. 

Again this should have been discussion - but code makes it easier to reply so it is good as well